src/wp-includes/comment.php

```diff
@@ -2429,9 +2429,17 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 
 	$maybe_notify = $is_note ? get_option( 'wp_notes_notify' ) : get_option( 'comments_notify' );
 
+	// By default, only notify for approved comments and notes.
+	if (
+		! isset( $comment->comment_approved ) ||
+		( '1' !== $comment->comment_approved && ! $is_note ) ) {
+			$maybe_notify = false;
+	}
+
 	/**
-	 * Filters whether to send the post author new comment notification emails,
-	 * overriding the site setting.
+	 * Filters whether to send the post author new comment and note notification emails,
+	 * overriding the site settings and defaults. By default, notifications are sent for
+	 * all notes and for approved comments.
 	 *
 	 * @since 4.4.0
```

Comment:
```suggestion
	 * @since 4.4.0
	 * @since 6.9.0 Comment approval status is checked before this filter. 
```

Minor, but I think it's worth documenting the change for posterity.

Reply:
Thanks for the suggestion! I'll change this to 7.0 as I plan to hold off on this logic change so late in the release cycle.